app-config-react-native-transformer/src/index.ts

```diff
@@ -0,0 +1,29 @@
+import { loadValidatedConfig } from '@lcdev/app-config';
+import { relative } from 'path';
+import { Transformer, upstreamTransformer } from './upstream-transformer';
+
+/**
+ * Transform is called by Metro for each package resolved in the React Native project
+ */
+export const transform: Transformer['transform'] = async ({ src, filename, options }) => {
+  // Get the relative path to app-config's entry point
+  const appConfigPath = relative(
+    options.projectRoot,
+    require.resolve('@lcdev/app-config', {
+      // Search relative to the project's root
+      paths: [options.projectRoot],
+    }),
+  );
+
+  // If the current module isn't app-config, pass-through to the default upstream transformer
+  if (filename !== appConfigPath) {
+    return upstreamTransformer.transform({ src, filename, options });
+  }
+
+  // Parse config and overwrite app-config's entry point with exported config JSON
+  const { parsedNonSecrets } = await loadValidatedConfig();
+
+  const modifiedSrc = `module.exports = ${JSON.stringify(parsedNonSecrets)};`;
```

Comment:
I pondered this for webpack plugin - is there any reason not to just use fullConfig? If there are secrets, I'm sure the person is aware that they are in a frontend (or in this case a mobile app). Thing is that `parsedNonSecrets` won't include `APP_CONFIG` (which is treated as neither secret nor nonSecret).

Reply:
I've unfortunately seen real-world cases where developers are extending a monorepo root `.app-config` from their frontend project to re-use a variable, but in the process they are leaking a ton of config meant only for backend. This example was thankfully only non-secret, but worried that this could happen with secrets if you're not careful.

Of course, this would require explicitly extending the secrets file within your frontend.

Reply:
Either way we should be consistent between RN (Metro) and Webpack. I'll switch to use the same as webpack, as I agree with your point that it is opt-in.

Reply:
Just as a point here, wouldn't `additionalProperties: false` help?